File: backend/src/main/kotlin/bo/edu/umsa/backend/dto/TaskDashboardDto.kt
package bo.edu.umsa.backend.dto

data class TaskDashboardDto(
    val totalTasks: Int,
    val totalCompletedTasks: Int,
    val totalCompletedWithDelayTasks: Int,
    val totalInProgressTasks: Int,
    val totalPendingTasks: Int,
    val totalDelayedTasks: Int,
    val totalHighPriorityTasks: Int,
    val totalMediumPriorityTasks: Int,
    val totalLowPriorityTasks: Int,
    val taskByDate: List<TaskByDateDto>,
)

data class TaskByDateDto(
    val year: Int,
    val month: Int,
    val completedTasks: Int,
    val completedWithDelayTasks: Int,
    val inProgressTasks: Int,
    val pendingTasks: Int,
    val delayedTasks: Int,
)

File: frontend/src/app/features/home/models/task-dashboard.dto.ts
export interface TaskDashboardDto {
    totalTasks: number;
    totalCompletedTasks: number;
    totalCompletedWithDelayTasks: number;
    totalInProgressTasks: number;
    totalPendingTasks: number;
    totalDelayedTasks: number;
    totalHighPriorityTasks: number;
    totalMediumPriorityTasks: number;
    totalLowPriorityTasks: number;
    taskByDate: TaskByDateDto[];
}

export interface TaskByDateDto {
    year: number;
    month: number;
    completedTasks: number;
    completedWithDelayTasks: number;
    inProgressTasks: number;
    pendingTasks: number;
    delayedTasks: number;
}
